tests/test_solvers.py

```diff
@@ -422,3 +429,27 @@ def test_ca_certificates_pins():
                     break
             else:
                 raise AssertionError("ca-certificates not found in LINK actions")
+
+
+def test_python_update_should_not_uninstall_history():
+    """
+    https://github.com/conda/conda-libmamba-solver/issues/341
+    """
+    channels = "--override-channels", "-c", "defaults"
+    solver = "--solver", "libmamba"
+    with make_temp_env("python=3.11", "numpy", *channels, *solver) as prefix:
+        assert package_is_installed(prefix, "python=3.11")
+        assert package_is_installed(prefix, "numpy=*=py311*")
+        with pytest.raises(
+            LibMambaUnsatisfiableError,
+            match=r"numpy.|\n+python 3\.12.*is not installable",
+        ):
+            run_command(
+                Commands.INSTALL,
+                prefix,
+                "python=3.12",
```

Comment:
This won't be reproduce in the future when `defaults` provides numpy for Python 3.12. We need an alternative package that we know won't be available in a future version of Python.